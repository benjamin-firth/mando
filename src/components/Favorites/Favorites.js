import React from 'react';
import Character from '../Character/Character.js';
import PropTypes from 'prop-types';

const Favorites = ({ faves, updateFave }) => {
    if (faves.length === 0) {
      return <h1>Sorry, No Faves</h1>
    }
    let characters = faves.map(character => {
      return <Character {...character} isFave={true} updateFave={updateFave}/>
    });

    return (
      <section className="favorite">
        <h1>Your Faves:</h1>
          {characters}
      </section>
    )
}

Favorites.propTypes = {
  faves: PropTypes.array,
  updateFave: PropTypes.func
}

export default Favorites;
