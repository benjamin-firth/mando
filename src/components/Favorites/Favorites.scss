.favorite {
  height: 90vh;
  overflow: scroll;
}