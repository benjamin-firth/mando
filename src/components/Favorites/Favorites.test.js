import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './Favorites.js';

describe('Favorites', () => {
  it('should match the snapshot when there are no favorites', () => {
    let wrapper = shallow(<Favorites faves={[]}/>);

    expect(wrapper).toMatchSnapshot();
  })

  it('should match snapshot when there are favorites', () => {
    let wrapper = shallow(<Favorites faves={['favorite1, favorite2, favorite3']} />);

    expect(wrapper).toMatchSnapshot();
  })
})
