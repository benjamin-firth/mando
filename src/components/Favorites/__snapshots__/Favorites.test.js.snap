// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot when there are favorites 1`] = `
<section
  className="favorite"
>
  <h1>
    Your Faves:
  </h1>
  <Character
    0="f"
    1="a"
    10=" "
    11="f"
    12="a"
    13="v"
    14="o"
    15="r"
    16="i"
    17="t"
    18="e"
    19="2"
    2="v"
    20=","
    21=" "
    22="f"
    23="a"
    24="v"
    25="o"
    26="r"
    27="i"
    28="t"
    29="e"
    3="o"
    30="3"
    4="r"
    5="i"
    6="t"
    7="e"
    8="1"
    9=","
    isFave={true}
  />
</section>
`;

exports[`Favorites should match the snapshot when there are no favorites 1`] = `
<h1>
  Sorry, No Faves
</h1>
`;
